core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java

```diff
@@ -144,8 +148,19 @@ public static OAuthTokenResponse fetchToken(RESTClient client, Map<String, Strin
     Map<String, String> request = clientCredentialsRequest(
         credential, scope != null ? ImmutableList.of(scope) : ImmutableList.of());
 
+    Map<String, String> requestHeaders;
+    if (headers.containsKey(AUTHORIZATION_HEADER)) {
+      requestHeaders = headers;
+    } else {
+      String credentialAsBase64 = Base64.getEncoder().encodeToString(credential.getBytes(StandardCharsets.UTF_8));
+      requestHeaders = ImmutableMap.<String, String>builder()
+          .putAll(headers)
+          .put(AUTHORIZATION_HEADER, BASIC_PREFIX + credentialAsBase64)
```

Comment:
It seems like this will put the credential as a basic auth header, but `clientCredentialsRequest()` already puts it in the payload so it is really needed?

Reply:
I thought that you said we preferred to use the Basic header, so I added it. I didn't remove the ID and secret from the payload because it is used in some cases, like when there is a Bearer token that needs to be passed in Authorization for the catalog. We can do whatever you think is right here.

Reply:
I see, thanks for the context. If you need to keep it in the payload anyway, then I'd vote to not add the header also. The spec wording seemed more about avoiding putting it in the payload.